### PR TITLE
Fixes to FPS, following errors in examples

### PR DESCRIPTION
**Pull request by rosecers**
I re-ran the FPS example and discovered a few new typos/bugs. Should be fixed here.

### CODE REVIEW

**Review by Luthaf — COMMENTED** [review]
tests

**Comment by rosecers** [comment]
@Luthaf I misspoke,

image

only typos...

**Comment by Luthaf** [comment]
Yes, but this kind of typos would have been caught by tests, and this is a good opportunity to add some anyway =)

**Comment by rosecers** [comment]
> Yes, but this kind of typos would have been caught by tests, and this is a good opportunity to add some anyway =)

image

**Comment by rosecers** [comment]
> > Yes, but this kind of typos would have been caught by tests, and this is a good opportunity to add some anyway =)
> 
> image

This is me saying "You're right, and as such, I added more tests".

File: tests/selection.py [hunk 1]
```diff
@@ -99,20 +125,28 @@ def test_others_orthogonal(self):
...
+    def setUp(self):
         self.model = SampleFPS
 
+    def test_supplied_indices(self):
+        """
+        This test checks FPS will accept pre-defined indices.
+        """
+        X, Y = load_boston(return_X_y=True)
+        model = self.model(X=X, mixing=0.5, Y=Y, idxs=[0, 1])
+        model.select(n=2)
```
Comment (Luthaf) [inline tests/selection.py:137]: Could we also check which indexes are selected? As long as we pass an initial value, it should be deterministic.

File: tests/selection.py [hunk 2]
```diff
@@ -50,14 +50,42 @@ def test_bad_X(self):
...
+    def setUp(self):
         self.model = FeatureFPS
 
+    def test_supplied_indices(self):
+        """
+        This test checks FPS will accept pre-defined indices.
+        """
+        X, Y = load_boston(return_X_y=True)
+        model = self.model(X=X, mixing=0.5, Y=Y, idxs=[0, 1])
+        model.select(n=2)
```
Comment (Luthaf) [inline tests/selection.py:84]: Same, it would be nice to check which indexes are selected

**Review by Luthaf — COMMENTED** [review]
Looks good in general, but it seems that code coverage is broken (by my fault). I tried to fix it, please give me 15 min to finish checking this!

**Comment by Luthaf** [comment]
Ok, I'm done messing with this branch!

File: tests/selection.py (hunk 2 above)
Comment (rosecers) [reply tests/selection.py]: > Ok, I'm done messing with this branch!

 I added a few more tests, but it's not really realistic to check all the indices for sample selection, given that many samples are effectively equivalent and can be chosen randomly in place of one another. I did add it for FS, for which there are only 13 features.

image

File: tests/selection.py [hunk 3]
```diff
@@ -97,22 +136,41 @@ def test_others_orthogonal(self):
...
+    def setUp(self):
         self.model = SampleCUR
 
     def test_orthogonalize(self):
         """
         This test checks that the orthogonalization that occurs after each
         selection leads to a null sample row
         """
-        self.setup()
+        self.setUp()
```
Comment (Luthaf) [inline tests/selection.py:173]: shouldn't this function be called automatically?

File: tests/selection.py [hunk 4]
```diff
@@ -129,7 +187,7 @@ def test_others_orthogonal(self):
         This test checks that the orthogonalization that occurs after each
         selection leads to orthogonal sample rows
         """
-        self.setup()
+        self.setUp()
```
Comment (Luthaf) [inline tests/selection.py:190]: same with automatic call

**Review by Luthaf — APPROVED** [review]

File: tests/selection.py (hunk 3 above)
Comment (rosecers) [reply tests/selection.py:173]: Yes, it should, and was... fixing.

**Pull request merged**